libs/atqr/domain/src/lib/valueObjects/notification.ts

```diff
@@ -0,0 +1,114 @@
+import { ValueObject } from '../../utils/valueObject';
+import { Guid } from '@tokilabs/lang';
+import { NotificationLogEntry } from './notificationLogEntry.valueObject';
+import { MessageTemplateId } from '../services';
+
+enum NotificationCategory {
+  EuDuvidoProductUpdates = 'EuDuvidoProductUpdates',
+  EuDuvidoPromotions = 'EuDuvidoPromotions',
+  NewInvitations = 'NewInvitations',
+  ChallengeUpdates = 'ChallengeUpdates',
+}
+
+export enum NotificationChannel {
+  Email = 'Email',
+  SMS = 'SMS',
+  Push = 'Push',
+}
+
+export class Notification extends ValueObject<Notification> {
+  constructor(
+    public readonly id: Guid,
+    public readonly category: NotificationCategory,
+    public readonly templateId: MessageTemplateId,
+    public readonly allowedChannels: NotificationChannel[],
+    public readonly createdAt: Date,
+    public readonly requiresUserAction: boolean,
+    public readonly actionTaken: boolean,
+    public readonly contactLog: NotificationLogEntry[],
+    public readonly nextContactDate: Date,
+    public readonly waitingForUserAction: boolean,
```

Comment:
In Disl, when a parameter has a nested statement, it means it's a property that executes the nested statement and returns the result.

Also, not your mistake, but let's change this to `isWaitingForUserAction` and `setWaitingForUserAction`